src/zeep/xsd/visitor.py

```diff
@@ -802,10 +802,11 @@ def visit_union(self, node, parent):
         """
         # TODO
         members = node.get('memberTypes')
+        default_namespace = node.nsmap.get(None)
         types = []
         if members:
             for member in members.split():
-                qname = as_qname(member, node.nsmap, self.document._target_namespace)
+                qname = as_qname(member, node.nsmap, default_namespace or self.document._target_namespace)
```

Comment:
Ah yeah, you're right. Then leaving the self.document._target_namespace as arg should fix it perhaps. I can look at it later today